go/common/enclave.go

```diff
@@ -109,9 +109,8 @@ type Enclave interface {
 type BlockSubmissionResponse struct {
 	BlockHeader *types.Header // the header of the consumed block. Todo - only the hash required
 
-	ProducedRollup ExtRollup // The new Rollup when ingesting the block produces a new Rollup
```

Comment:
I've renamed these fields to clarify what they mean, and dropped the `FoundNewHead` method (you can see if you've found a new head because the `NewRollupHeader` will be non-nil.